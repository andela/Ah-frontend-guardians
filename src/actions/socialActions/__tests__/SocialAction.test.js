import configureMockStore from 'redux-mock-store';
import React from 'react';
import thunk from 'redux-thunk';
import mockAxios from 'jest-mock-axios';
import facebookLoginAction from '../facebookAction';
import googleLoginAction from '../googleAction';
import "babel-polyfill";
import { shallow } from 'enzyme';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});
let postFunc = (myfunction) => {
  mockAxios.post('MockURL', { access_token: 'accessToken' });
  store.dispatch(myfunction({ access_token: 'accessToken' }));
  expect(store.getActions()).toEqual([]);
};

describe('Social Actions', () => {
  afterEach = () => {
    mockAxios.reset();
  };

  it('dispatch actions for facebook login', () => {
    postFunc(facebookLoginAction)
  });

  it('dispatch actions for google login', () => {
    postFunc(googleLoginAction)
  });

  it('dispatch actions for google login', () => {
    let wrapper = shallow(<facebookLoginAction />);
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatch actions for google login', () => {
    let wrapper = shallow(<googleLoginAction />);
    expect(wrapper).toMatchSnapshot();
  });

});
