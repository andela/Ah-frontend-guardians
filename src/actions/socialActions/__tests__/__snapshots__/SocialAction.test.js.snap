// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Actions dispatch actions for google login 1`] = `ShallowWrapper {}`;

exports[`Social Actions dispatch actions for google login 2`] = `ShallowWrapper {}`;
